src/object/gradient.hpp

```diff
@@ -59,11 +65,87 @@ class Gradient final :
   GradientDirection get_direction() const { return m_gradient_direction; }
   std::string get_direction_string() const;
   void set_direction(const GradientDirection& direction);
-  void set_direction(const std::string& direction);
 
   void set_layer(int layer) { m_layer = layer; }
   int get_layer() const { return m_layer; }
 
+  /**
+   * @scripting
+   * @description Sets the direction of the gradient.
+   * @param string $direction Can be "horizontal", "vertical", "horizontal_sector" or "vertical_sector".
+   */
+  void set_direction(const std::string& direction);
+#ifdef DOXYGEN_SCRIPTING
+  /**
+   * @scripting
+   * @description Returns the direction of the gradient.
+                  Possible values are "horizontal", "vertical", "horizontal_sector" or "vertical_sector".
+   */
+  std::string get_direction() const;
+#endif
```

Comment:
Because in reality, `Gradient::get_direction_string` is used, instead of `Gradient::get_direction`, because `GradientDirection Gradient::get_direction()` already exists and there would be ambiguity.

This is done so the function properly appears in the documentation.